src/main/java/StringManipulation.java

```diff
@@ -1,32 +1,114 @@
-public class StringManipulation implements StringManipulationInterface {
+/* Chris Nevares
+ * 6/11/23
+ * CS410, Spring 2023
+ * Junit Testing Assignment - StringManipulationTest
+ *
+ * The StringManipulation class provides various ways to manipulate an input string
+ * Most methods deal with the number of words in the input string.
+ *
+ */
+
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
+public class StringManipulation implements StringManipulationInterface {
+    private String myString = null;
     @Override
+
     public String getString() {
-        return null;
+        return this.myString;
     }
 
     @Override
     public void setString(String string) {
+
+        this.myString = string;
     }
 
     @Override
     public int count() {
-        return 0;
+        if(myString == null) {
+            throw new NullPointerException("String is null.");
+        }
+        String regex = "\\w*['-]?\\w+";
+        Pattern pattern = Pattern.compile(regex);
+        Matcher matcher = pattern.matcher(myString);
+        int wordCount = 0;
+        while(matcher.find()) {
+            wordCount++;
+        }
+
+        return wordCount;
     }
 
     @Override
-    public String removeNthCharacter(int n, boolean maintainSpacing) {
-        return null;
+    public String removeNthCharacter(int n, boolean maintainSpacing)  {
+        if(myString == null) {
+            throw new NullPointerException("String is null.");
```

Comment:
I wasn't able to think of the null case at first, nice implementation to cover the null case.